compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala

```diff
@@ -501,38 +501,47 @@ case class PortCppWriter (
   }
 
   private def getOutputPortFunctionMembers: List[CppDoc.Class.Member] = {
-    val paramNames = paramList.map(_._1).mkString(", ")
+    val invokeCall = s"this->m_port->invoke(${paramList.map(_._1).mkString(", ")});"
     val invokeBody = data.returnType match {
       case Some(_) => lines(
-        s"return this->m_port->invoke($paramNames);"
-      )
-      case None => lines(
-        s"""|if (this->m_port) {
-            |  this->m_port->invoke($paramNames);
-            |#if FW_PORT_SERIALIZATION
-            |} else if (this->m_serPort) {
-            |  Fw::SerializeStatus _status;
-            |  ${name}PortBuffer _buffer;
+        s"""|
+            |FW_ASSERT(this->m_port != nullptr);
+            |return $invokeCall
             |"""
-      ) ++
+      )
+      case None => List(
+        lines(
+          s"""|
+              |#if FW_PORT_SERIALIZATION
+              |FW_ASSERT((this->m_port != nullptr) || (this->m_serPort != nullptr));
+              |
+              |if (this->m_port != nullptr) {
+              |  $invokeCall
+              |} else {
```

Comment:
Can we move the else statement to the next line?

Reply:
Ok, fixed!